test/object-store/sync/flx_migration.cpp

```diff
@@ -84,6 +85,206 @@ static std::vector<ObjectId> fill_test_data(SyncTestFile& config, std::optional<
+    REQUIRE(*migration_store->get_migrated_partition() == migrated_partition);
+
+    // Verify conversion from PBS to FLX sync config - different partition value will not cause
+    // LogicError until migration is complete
+    REQUIRE_NOTHROW(migration_store->convert_sync_config(sync_config));
+    auto migrated_config = migration_store->convert_sync_config(sync_config);
+    REQUIRE(sync_config != migrated_config);
+    REQUIRE(migrated_config->flx_sync_requested);
+    // Verify no conversion from FLX to FLX sync config
+    sync_config->flx_sync_requested = true;
+    sync_config->partition_value = {};
+    REQUIRE_NOTHROW(migration_store->convert_sync_config(sync_config));
+    migrated_config = migration_store->convert_sync_config(sync_config);
+    REQUIRE(sync_config == migrated_config);
+    REQUIRE(migrated_config->flx_sync_requested);
+}
+
+static void check_migration_complete(const std::shared_ptr<sync::MigrationStore>& migration_store)
+{
+    auto sync_config = std::make_shared<SyncConfig>(nullptr, "some_other_parition");
+    REQUIRE_FALSE(sync_config->flx_sync_requested);
+
+    REQUIRE(migration_store->is_migrated());
+    REQUIRE_FALSE(migration_store->is_migration_in_progress());
+    REQUIRE(migration_store->get_query_string());
+    REQUIRE(*migration_store->get_query_string() == rql_string);
+    REQUIRE(migration_store->get_migrated_partition());
+    REQUIRE(*migration_store->get_migrated_partition() == migrated_partition);
+
+    // Verify logic error is thrown if partition value does not match migrated partition
+    CHECK_THROWS_AS(migration_store->convert_sync_config(sync_config), LogicError);
+    // Verify conversion from PBS to FLX sync config with matching partition values
+    sync_config->partition_value = migrated_partition;
+    REQUIRE_NOTHROW(migration_store->convert_sync_config(sync_config));
+    auto migrated_config = migration_store->convert_sync_config(sync_config);
+    CHECK(sync_config != migrated_config);
+    REQUIRE(migrated_config->flx_sync_requested);
+    // Verify no conversion from FLX to FLX sync config
+    sync_config->flx_sync_requested = true;
+    sync_config->partition_value = {};
+    REQUIRE_NOTHROW(migration_store->convert_sync_config(sync_config));
+    migrated_config = migration_store->convert_sync_config(sync_config);
+    REQUIRE(sync_config == migrated_config);
+    REQUIRE(migrated_config->flx_sync_requested);
+}
+
+static void check_subscription(const sync::SubscriptionSet& sub_set, const std::string& table_name,
+                               const std::string& query_string)
+{
+    auto sub_name = std::string("flx_migrated_" + table_name);
+    auto table_sub = sub_set.find(sub_name);
+    REQUIRE(table_sub);
+    REQUIRE(table_sub->query_string == query_string);
+    REQUIRE(table_sub->object_class_name == table_name);
+    REQUIRE(table_sub->name == sub_name);
+}
+
+TEST_CASE("Migration store", "[flx][migration]") {
```

Comment:
since these are not integration tests and are logically separated from the rest of the tests, maybe they should be in their own file (i.e, migration_store_test.cpp)